js/packages/cli/src/helpers/instructions.ts

```diff
@@ -218,14 +219,25 @@ export async function createConfigAccount(
     4 +
     Math.ceil(configData.maxNumberOfLines.toNumber() / 8);
 
+  let rent = await anchorProgram.provider.connection.getMinimumBalanceForRentExemption(
+    size,
+  );
+  
+  // calculate rent
+  if (typeof rentTimeInDays !== 'undefined') {
+    console.log('Instead of paying ' + rent + ' Lamports for rent')
+    let rentMultiplier = rentTimeInDays / 30;
+    rent = rent / 24 * rentMultiplier;
```

Comment:
where are you getting this math from?

Reply:
Hi Jordan,
you are right, it is not correct. Looks like i commited a WIP. 
What we really would need to do is
// First get the number of epochs. +1 to be save in case the epoch ends in the next minutes.
const epochs = Math.ceil(rentTimeInDays/2) + 1;

// get the rent per byte. There is no implementation in web3 for it right now. To get an approximation divide the cost of rent exemption with the appoximate number of epochs in two years
const rent_per_byte_epoch = rent / 365;

// and finally multiply our count of epochs with the rent.
rent = epochs * rent_per_byte_epoch

Reply:
We can also use the calculation described in the docs:
> As of writing, the fixed rent fee is 19.055441478439427 lamports per byte-epoch on the testnet and mainnet-beta clusters. An epoch is targeted to be 2 days (For devnet, the rent fee is 0.3608183131797095 lamports per byte-epoch with its 54m36s-long epoch).
>
> This value is calculated to target 0.01 SOL per mebibyte-day (exactly matching to 3.56 SOL per mebibyte-year):
>
> ```text
> Rent fee: 19.055441478439427 = 10_000_000 (0.01 SOL) * 365(approx. day in a year) / (1024 * 1024)(1 MiB) / (365.25/2)(epochs in 1 year)
> ```
> And rent calculation is done with the f64 precision and the final result is truncated to u64 in lamports.

Or do it how the official `solana rent` command does it:
([source](https://github.com/solana-labs/solana/blob/3c5f505d3e3ded8b3110b44bc458d01701b93e4d/cli/src/cluster_query.rs#L2120-L2135))
```rs
    let epoch_schedule = rpc_client.get_epoch_schedule()?;
    let rent_account = rpc_client.get_account(&sysvar::rent::id())?;
    let rent: Rent = rent_account.deserialize_data()?;
    let rent_exempt_minimum_lamports = rent.minimum_balance(data_length);
    let seconds_per_tick = Duration::from_secs_f64(1.0 / clock::DEFAULT_TICKS_PER_SECOND as f64);
    let slots_per_year =
        timing::years_as_slots(1.0, &seconds_per_tick, clock::DEFAULT_TICKS_PER_SLOT);
    let slots_per_epoch = epoch_schedule.slots_per_epoch as f64;
    let years_per_epoch = slots_per_epoch / slots_per_year;
    let (lamports_per_epoch, _) = rent.due(0, data_length, years_per_epoch);
    let cli_rent_calculation = CliRentCalculation {
        lamports_per_byte_year: rent.lamports_per_byte_year,
        lamports_per_epoch,
        rent_exempt_minimum_lamports,
        use_lamports_unit,
    };
```